python/tests/test_events.py

```diff
@@ -0,0 +1,366 @@
+    model_module = importlib.import_module(model_name)
+
+    model = model_module.getModel()
+    return model
+
+
+def create_sbml_model(
+        initial_assignments,
+        parameters,
+        rate_rules,
+        species,
+        events,
+        to_file: str = None,
+):
+    """Create an SBML model from simple definitions.
+
+    See the model definitions and usage in :py:func:`model` for example input.
+
+    The default initial concentration of species is `1.0`. This can currently
+    be changed by specifying an initial assignment.
+    """
+    document = libsbml.SBMLDocument(3, 1)
+    model = document.createModel()
+
+    compartment = model.createCompartment()
+    compartment.setId('compartment')
+    compartment.setConstant(True)
+    compartment.setSize(1)
+    compartment.setSpatialDimensions(3)
+    compartment.setUnits('dimensionless')
+
+    for species_id in species:
+        species = model.createSpecies()
+        species.setId(species_id)
+        species.setCompartment('compartment')
+        species.setConstant(False)
+        species.setSubstanceUnits('dimensionless')
+        species.setBoundaryCondition(False)
+        species.setHasOnlySubstanceUnits(False)
+        species.setInitialConcentration(1.0)
+
+    for target, formula in initial_assignments.items():
+        initial_assignment = model.createInitialAssignment()
+        initial_assignment.setSymbol(target)
+        initial_assignment.setMath(libsbml.parseL3Formula(formula))
+
+    for target, formula in rate_rules.items():
+        rate_rule = model.createRateRule()
+        rate_rule.setVariable(target)
+        rate_rule.setMath(libsbml.parseL3Formula(formula))
+
+    for parameter_id, parameter_value in parameters.items():
+        parameter = model.createParameter()
+        parameter.setId(parameter_id)
+        parameter.setConstant(True)
+        parameter.setValue(parameter_value)
+        parameter.setUnits('dimensionless')
+
+    for event_id, event_def in events.items():
+        event = model.createEvent()
+        event.setId(event_id)
+        event.setName(event_id)
+        event.setUseValuesFromTriggerTime(True)
+        trigger = event.createTrigger()
+        trigger.setMath(libsbml.parseL3Formula(event_def['trigger']))
+        trigger.setPersistent(True)
+        trigger.setInitialValue(True)
+        assignment = event.createEventAssignment()
+        assignment.setVariable(event_def['target'])
+        assignment.setMath(libsbml.parseL3Formula(event_def['assignment']))
+
+    if to_file:
+        libsbml.writeSBMLToFile(
+            document,
+            str(to_file),
+        )
+
+    # Need to return document, else AMICI throws an error.
+    # (possibly due to garbage collection?)
+    return document, model
+
+
+def get_model_definition(model_name):
+    if model_name == 'events_plus_heavisides':
+        return model_definition_events_plus_heavisides()
+#    elif model_name == 'simple_event':
```

Comment:
don't comment this out?

Reply:
That test case isn't yet implemented. I wonder whether we need it at all, as the above one really covers many weird cases... I might rather implement another case with nested events. This would tests some further functionality.

Reply:
fine, I thought there might have already been an implementation somewhere else. Simply delete in that case.